cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
I didn't modify the vSphere kube-up code yet. This change doesn't break it (DNS doesn't work there), but it's an incomplete fix. Another change will be needed to get the UI and DNS to work with the vSphere kube-up. I plan to do that in the next week or so after I get a couple other things cleared up.